src/services/RPCSelector.ts

```diff
@@ -0,0 +1,69 @@
+import { JsonRpcProvider, Block } from '@ethersproject/providers';
+import { isTimestampMoreRecentThan } from '../utils/helpers';
+
+interface ProviderComparand {
+  blockNumber: number;
+  url: string;
+}
+
+class RPCSelector {
+  readonly urls: string[];
+  readonly preferredProviderUrl: string;
+  readonly rpcRequestTimeout: number;
+  readonly maxBlockAge: number;
+
+  constructor(urls: string | string[], rpcRequestTimeout = 15000, maxBlockAge = 60) {
```

Comment:
BTW, I was thinking of making this an object instead of bloating the constructor with arguments.
What do you guys think?

Reply:
can we make preferredProviderUrl() a method?

Reply:
I don't see any difference in setting it as a property or making it a method. It serves the same purpose.